src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -340,6 +340,15 @@ private void OnFileSystemEntryChange(string fullPath)
         {
             try
             {
+                // this can happen when the fullPath matches the root directory path
+                // but the root path is always created with a trailing slash
+                // this was masked by a side effect (bug) in RenamedEventArgs that was appending the trailing slash to fullPath value
+                // that behavior was changed with the fix for https://github.com/dotnet/runtime/issues/62606
+                if (fullPath.Length < _root.Length)
+                {
+                    return;
+                }
```

Comment:
Can you please ensure `Renamed/FileSystemEventArgs.FullPath` returns with a trailing separator if the `[old]Name` argument is null or empty. 
I'm sure we want to keep that specific part of the old behavior.

Let me know if you need help on doing that and I can give you some pointers.

Reply:
Fixed in commit aaf4b45

Reply:
@Jozkee this was the final item on the review list. Tests pass and the previous failures did not seem related and could have been transient issues. Thanks!